### PR TITLE
examples: Add a example showing how to switch to offboard mode in PX4

### PR DESCRIPTION
**Pull request by zehortigoza**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -161,6 +161,10 @@ noinst_PROGRAMS += heartbeat-print
 heartbeat_print_SOURCES = \
 	examples/heartbeat-print.cpp
 
+noinst_PROGRAMS += offboard-mode
+offboard_mode_SOURCES = \
+	examples/offboard-mode.cpp
```
Comment (lucasdemarchi) [inline Makefile.am]: since this is px4-specific, it shouldn't be named px4-offboard-mode.cpp

File: examples/offboard-mode.cpp [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    mavlink_command_long_t cmd = {};
+
+    cmd.command = MAV_CMD_DO_SET_MODE;
+    cmd.target_system = TARGET_SYSTEM_ID;
+    cmd.target_component = MAV_COMP_ID_ALL;
+
+    switch (mode) {
+    case PX4_MODE_OFFBOARD:
+        cmd.param1 = 1;// custom mode
+        cmd.param2 = 6;// offboard
```
Comment (lucasdemarchi) [inline examples/offboard-mode.cpp]: make these 2 defines so you don't have to add the comment

File: examples/offboard-mode.cpp [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    desc[1].fd = timeout_fd;
+    desc[1].events = POLLIN | POLL_ERROR_EVENTS;
+    desc[1].revents = 0;
+
+    while (!g_should_exit) {
+        int ret = poll(desc, sizeof(desc) / sizeof(struct pollfd), -1);
+
+        if (ret < 1) {
+            fprintf(stderr, "poll returned a error (%m)\n");
+            continue;
```
Comment (lucasdemarchi) [inline examples/offboard-mode.cpp]: you never break?

File: examples/offboard-mode.cpp [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    if (timeout_fd < 0) {
+        fprintf(stderr, "Unable to create timerfd (%m)\n");
+        goto timerfd_error;
+    }
+
+    ts.it_interval.tv_sec = 0;
+    ts.it_interval.tv_nsec = 1000000 * 50;// 50msec
+    ts.it_value.tv_sec = ts.it_interval.tv_sec;
+    ts.it_value.tv_nsec = ts.it_interval.tv_nsec;
+    timerfd_settime(timeout_fd, 0, &ts, NULL);
```
Comment (lucasdemarchi) [inline examples/offboard-mode.cpp]: see src/common/util.h to use the macros.

File: examples/offboard-mode.cpp [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+static void position_set_send(float x, float y, float z, float yaw)
+{
+    mavlink_message_t msg;
+    mavlink_set_position_target_local_ned_t set_position = {};
+    timespec now;
+    uint64_t now_ms;
+    static uint64_t initial_ms = 0;
+
+    clock_gettime(CLOCK_MONOTONIC, &now);
+    now_ms = (now.tv_nsec / 1000000) + (now.tv_sec * 1000);
```
Comment (lucasdemarchi) [inline examples/offboard-mode.cpp]: there's  now_usec() in src/common/util.h... you can use it.

Comment (lucasdemarchi) [inline examples/offboard-mode.cpp]: s/ms/msec/ for consistency.

**Comment by lucasdemarchi** [comment]
A comment on top of the file stating what it does would be good... maybe later we can add a way for the user to interact with it to move the drone around.

**Comment by lucasdemarchi** [comment]
right now as far as I understood it just change the mode to offboard and set home position by sending all zeros.

**Comment by zehortigoza** [comment]
New version with the requested changes.

**Comment by lucasdemarchi** [comment]
Applied, thanks.

I added a commit on top splitting the setup phase from the main function a little bit.

**Pull request closed without merging**